spec/services/stats/fee_scheme_usage_generator_spec.rb

```diff
@@ -116,35 +115,35 @@ def amend_claim(claim, case_type_id, case_type_name)
       expect(csv.headers).to eq(expected_headers)
     end
 
-    context 'when generating all month sections' do
+    context 'when generating the section 5 months ago' do
       subject(:call) { described_class.new.call }
 
-      it 'returns rows containing the correct numbers of total claims' do
-        expect(csv['Total number of claims']).to eq(total_claims_array)
+      it 'correctly populates the LGFS 9 row' do
+        expect(find_row(5.months.ago.strftime('%B'), 'LGFS 9')['Total number of claims']).to eq('1')
       end
     end
 
     context 'when generating the most recent month' do
       it 'returns rows containing the correct fee schemes' do
-        expect(csv['Fee scheme'][50...59]).to eq(fee_scheme_array)
-      end
-
-      it 'returns rows containing the correct total value of claims' do
-        expect(csv['Total value of claims'][50...60]).to eq([
-                                                              '380.0', '0', '0', '0', '0.0', '0', '0', '75.02', '0', nil
-                                                            ])
+        expect(csv['Fee scheme'][(block_length * 5)..-2]).to eq(fee_scheme_array)
```

Comment:
This is a bit opaque. It's not clear what `[(block_length * 5)..-2]` really means. I kind of get that it's finding the last block of fee scheme names. But it's not that easy to interpret without context.

Could you could do something like `expect(csv['Fee scheme'].uniq.compact).to eq(fee_scheme_array)`? It produces (I think) the same result and is a bit more idiomatic.

Reply:
Good idea, thanks Mark! That's much neater and still tests the same thing. I've changed it to this